python_bindings/stub/CMakeLists.txt

```diff
@@ -1,12 +1,17 @@
-if (NOT TARGET Halide::PyStubs)
-    add_library(Halide_PyStubs STATIC ${CMAKE_CURRENT_LIST_DIR}/PyStubImpl.cpp)
-    add_library(Halide::PyStubs ALIAS Halide_PyStubs)
+add_library(Halide_PyStubs STATIC ${CMAKE_CURRENT_LIST_DIR}/PyStubImpl.cpp)
+add_library(Halide::PyStubs ALIAS Halide_PyStubs)
 
-    target_link_libraries(Halide_PyStubs PRIVATE pybind11::module)
-    target_link_libraries(Halide_PyStubs PUBLIC Halide::Halide)
-    set_target_properties(Halide_PyStubs PROPERTIES
-                          EXPORT_NAME PyStubs
-                          CXX_VISIBILITY_PRESET hidden
-                          VISIBILITY_INLINES_HIDDEN TRUE
-                          POSITION_INDEPENDENT_CODE ON)
-endif ()
+target_link_libraries(Halide_PyStubs PRIVATE Halide::Halide)
+
+# Don't add a dependency on pybind11::pybind11 here: that will add
+# a phantom dependency is propagated into our install setup.
+# All we really need here is a path to the include directories for pybind11,
+# which is publicly documented as `pybind11_INCLUDE_DIRS`,
+target_include_directories(Halide_PyStubs PRIVATE "${HALIDE_INCLUDES}" "${pybind11_INCLUDE_DIRS}")
```

Comment:
There's a better solution for this: link to `$<BUILD_INTERFACE:pybind11::pybind11>` in the `target_link_libraries()` command above. Then re-word the comment to explain why the dependency disappears post-build.

Reply:
Cool, let me try that